java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -221,12 +222,31 @@ public final void visitArrayAccessExpression(ArrayAccessExpressionTree tree) {
     @Override
     public final void visitAssignmentExpression(AssignmentExpressionTree tree) {
       evaluateExpression(currentState, tree.variable());
-      evaluateExpression(currentState, tree.expression());
+      SymbolicBooleanConstraint result = evaluateExpression(currentState, tree.expression());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "result" which hides the field declared at line 79. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)